dev/Pager/PrototypePager/PrototypePager.cs

```diff
@@ -52,44 +75,46 @@ public enum ButtonVisibilityMode { Auto, AlwaysVisible, HiddenOnEdge, None, }
         private int PreviousPageIndex = -1;
 
         public event TypedEventHandler<PrototypePager, PageChangedEventArgs> PageChanged;
-        
+
         public PrototypePager()
         {
             this.DefaultStyleKey = typeof(PrototypePager);
-            this.Loaded += (s, args) => { SetValue(TemplateSettingsProperty, new PagerTemplateSettings(this)); };
+            this.Loaded += OnLoad;
+        }
 
+        private void OnLoad(object sender, RoutedEventArgs args)
+        {
+            // Attach Callbacks for property changes
+            RegisterPropertyChangedCallback(NumberOfPagesProperty, (s, e) => { OnNumberOfPagesChanged(); });
```

Comment:
You can actually set these up at the time that you register the dependency properties, by adding the callback to the PropertyMetadata object.

The code as it is has problems, since the Loaded event can fire multiple times if the same control gets removed and added back into the tree.

Reply:
Done